mesonbuild/environment.py

```diff
@@ -635,7 +635,9 @@ def create_new_coredata(self, options):
         self.coredata.meson_command = mesonlib.meson_command
         self.first_invocation = True
 
-    def is_cross_build(self) -> bool:
+    def is_cross_build(self, when_building_for: MachineChoice = MachineChoice.HOST) -> bool:
+        if when_building_for == MachineChoice.BUILD:
+            return False
```

Comment:
This is wrong and splits the logic again. The logic must be inside `coredata.is_cross_build` and it is _exactly_ whether the user has defined a cross file or not. We have documented that publicly. What is the problem you are trying to work around by having this extra logic here?

Reply:
> This is wrong and splits the logic again.

Oops, sorry, I misunderstood the architecture.

> What is the problem you are trying to work around by having this extra logic here?

The reason I assumed there would be a need to differentiate is that (unless I'm mistaken) Meson may create up to two instances of each compiler, one instance for the build machine and one for the host machine. I assumed that the instance for the build machine should be constructed with `is_cross=False` (for the compilers that have `is_cross`). The reason I assumed this is that implementations may choose to use the `exe_wrapper` when `self.is_cross == True`, and it seems weird to build a binary for the build machine and try to run it using the configured wrapper (if set).